### PR TITLE
Ensure that random seeds are synced across sharding ranks

### PR DESCRIPTION
**Pull request by PhilipVinc**
@lgravina1997 has recently noticed that we do not generate the same RNG seeds across hosts when running with multi-process  sharding.

(how come you never noticed @inailuig ?)

Anyhow, this PR fixes it.

Unsure how to test..

### CODE REVIEW

**Comment by PhilipVinc** [comment]
@inailuig this is not reproducibility! This is plain out wrong. If someone does not set a seed a randomly initialised network will have different weights on every rank.

**Comment by PhilipVinc** [comment]
@inailuig is this good now?

File: netket/jax/_utils_random.py [hunk 1]
```diff
@@ -30,16 +30,29 @@ def PRNGKey(
     The same seed will be distributed to all processes.
     """
     if seed is None:
-        key = jax.random.PRNGKey(random_seed())
-    elif isinstance(seed, int):
+        seed = random_seed()
+
+    if isinstance(seed, int):
```
[the remark this answers is not in the document]
Comment (PhilipVinc) [reply netket/jax/_utils_random.py:35]: I'd rather not have this assumption. 
Because if someone is generating a seed using time.time or numpy the program is identical but the output is not.

The always stated purpose of `nkjax.PRNGKey` is to ensure a common seed everywhere regardless of what you do, and it costs us little to keep this guarantee.

[the remark this answers is not in the document]
Comment (PhilipVinc) [reply netket/jax/_utils_random.py:35]: > I'm not a big fan of all of this babysitting, but ok. I would expect people knew what they are doing....

I'd like to start doing less babysitting, so I agree with you on this.

If you have a good argument for this case I'll listen! 
But nkjax.PRNGKey is really just there to ensure this...
